### PR TITLE
Add Analysis of pingSender effects on crash ping client delays

### PR DESCRIPTION
**Pull request by chutten**
It turns out that `pingSender` speeds things up. Nice.

### CODE REVIEW

File: projects/crash_ping_delays_pingSender.kp/knowledge.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+updated_at: 2017-03-07 13:25:46.782837
+tldr: How long does it take before we get crash pings from users that have pingSender
+  vs users who don't?
+thumbnail: images/output_26_1.png
+---
+### Crash Ping Submission and Recording Delays - pingSender
+
+This is follow-up analysis to the [Crash Ping Submission and Recording Delays by Channel](http://reports.telemetry.mozilla.org/post/projects/crash_ping_delays.kp) analysis previously performed.
+
+Specifically, this one investigates the difference between typical values of "recording delay" and "submission delay" before and after [pingSender started sending pings]()https://bugzilla.mozilla.org/show_bug.cgi?id=1310703.
```
Comment (harterrt) [inline projects/crash_ping_delays_pingSender.kp/knowledge.md]: Please fix the link formatting here.

File: projects/crash_ping_delays_pingSender.kp/knowledge.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+    {'application/channel': u'nightly',
+     'creationDate': u'2017-02-24T12:28:08.998Z',
+     'id': u'9cf083d2-c1cc-4a66-947c-ba4a783f9dce',
+     'meta/Date': u'Fri, 24 Feb 2017 12:28:11 GMT',
+     'meta/Timestamp': 1487939292673567488L,
+     'meta/creationTimestamp': 1.4879392889979999e+18,
+     'payload/crashDate': u'2017-02-24',
+     'payload/processType': u'main',
+     'pre': 'post'}
+
```
Comment (harterrt) [inline projects/crash_ping_delays_pingSender.kp/knowledge.md]: Probably being overly cautious, but please consider removing this block.

File: projects/crash_ping_delays_pingSender.kp/knowledge.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+    .reduceByKey(lambda a, b: a if a["meta/Timestamp"] < a["meta/Timestamp"] else b)\
+    .map(lambda pair: pair[1])
+```
+
+```python
+combined_count = combined.count()
+combined_deduped_count = combined_deduped.count()
+print "Filtered {} of {} crash pings ({:.2f}%)".format(combined_count - combined_deduped_count, combined_count, 100.0 * (combined_count - combined_deduped_count) / combined_count)
+```
+    Filtered 9684 of 47912 crash pings (20.21%)
```
Comment (harterrt) [inline projects/crash_ping_delays_pingSender.kp/knowledge.md]: Can you add a note summarizing the cumulative impact of the deduplication and main process filtering? E.g. pingSender affects ~20% of all crash pings.

Comment (chutten) [inline projects/crash_ping_delays_pingSender.kp/knowledge.md]: That's not the purpose of this analysis, though. And pingSender duplicates a much higher %ge than that, but only of main crash pings.

Comment (harterrt) [inline projects/crash_ping_delays_pingSender.kp/knowledge.md]: Ah, I misunderstood. No need to revise, thanks.

**Comment by chutten** [comment]
That ought to do it.

**Comment by harterrt** [comment]
LGTM! Thanks

**Pull request merged**